backend/src/main/java/ai/verta/modeldb/versioning/blob/diff_factory/BlobDiffFactory.java

```
@@ -0,0 +1,124 @@
+    // TODO: - contain the `#` then this functionality will break.
+    final Builder builder = BlobDiff.newBuilder()
+        .addAllLocation(Arrays.asList(location.split("#")));
+    builder.setStatus(DiffStatus.ADDED);
+    add(builder);
+    return builder.build();
+  }
+
+  public BlobDiff delete(String location) {
+    // TODO: Here used the `#` for split the locations but if folder
+    // TODO: - contain the `#` then this functionality will break.
+    final Builder builder = BlobDiff.newBuilder()
+        .addAllLocation(Arrays.asList(location.split("#")));
+    builder.setStatus(DiffStatus.DELETED);
+    delete(builder);
+    return builder.build();
+  }
+
+  public static BlobDiffFactory create(BlobExpanded blobExpanded) {
+    switch (blobExpanded.getBlob().getContentCase()) {
+      case ENVIRONMENT:
+        return new EnvironmentBlobDiffFactory(blobExpanded);
+      case CODE:
+        return new CodeBlobDiffFactory(blobExpanded);
+      case CONFIG:
+        return new ConfigBlobDiffFactory(blobExpanded);
+      case DATASET:
+        return new DatasetBlobDiffFactory(blobExpanded);
+      default:
+        return new BlobDiffFactory(blobExpanded) {
+          @Override
+          public List<BlobDiff> compare(BlobDiffFactory blobDiffFactory, String location) {
+            return Collections.emptyList();
+          }
+
+          @Override
+          public BlobDiff add(String location) {
+            return BlobDiff.newBuilder().build();
+          }
+
+          @Override
+          public BlobDiff delete(String location) {
+            return BlobDiff.newBuilder().build();
+          }
+
+          @Override
+          protected boolean typeEqual(BlobDiffFactory blobDiffFactory) {
+            return false;
+          }
+
+          @Override
+          protected void add(Builder builder) {
+          }
+
+          @Override
+          protected void delete(Builder builder) {
+          }
+        };
```

Instead of this, I prefer to throw Exception with the `Invalid blob case found` something like that.

---

fixed